### PR TITLE
Added `NSPipe` implementation

### PR DESCRIPTION
**Pull request by terhechte**
The implementation itself is very simple. In the unlikely event that the
`pipe` system call fails, this implementation will stop with a fatalError.
This seems to be the canonical way of doing it based on my cursory reading of
the rest of foundation.

I've also added a first, simple, unit test to make sure the pipe works as expected.

### CODE REVIEW

File: TestFoundation/TestNSPipe.swift [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+class TestNSPipe : XCTestCase {
+    
+    func test_NSPipe() {
+        let expectation = self.expectationWithDescription("Should read data")
+        let aPipe = NSPipe()
+        
+        let text = "test-pipe"
+        
+        dispatch_async(dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_BACKGROUND, 0)) { () -> Void in
```
Comment (phausler) [inline TestFoundation/TestNSPipe.swift]: the swift corelibs dispatch is not yet available for linux unfortunately so a thread or some other mechanism would have to be used to asynchronously read.

**Comment by terhechte** [comment]
I've updated the test to remove the libdispatch requirement.

File: TestFoundation/TestNSPipe.swift [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+import Foundation
+import XCTest
+#else
+import SwiftFoundation
+import SwiftXCTest
+#endif
+
+
+class TestNSPipe : XCTestCase {
+
```
Comment (phausler) [inline TestFoundation/TestNSPipe.swift]: This needs a method for allTests
```swift
public var allTests: [(String, () -> ())] { 
    return [
         ("test_NSPipe", test_NSPipe)
    ]
 }
```

File: TestFoundation/TestNSPipe.swift [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+
+class TestNSPipe : XCTestCase {
+    
+    func test_NSPipe() {
+        let aPipe = NSPipe()
+        let text = "test-pipe"
+        
+        // First write some data into the pipe
+        aPipe.fileHandleForWriting.writeData(text.dataUsingEncoding(NSUTF8StringEncoding)!)
```
Comment (phausler) [inline TestFoundation/TestNSPipe.swift]: This is relying on implicit bridging and should be 
```swift
text.bridge().dataUsingEncoding(NSUTF8StringEncoding)
```

**Comment by terhechte** [comment]
Thanks for these comments! I've updated the test and I've also included an additional assertion for the initial string to nsdata conversion.

**Comment by phausler** [comment]
One other minor thing it needs is the TestNSPipe.swift added to build.py (or rebase from master which now globs; which is preferable) and in TestFoundation/main.swift add the test to the XCTMain

**Comment by terhechte** [comment]
I did a rebase so that the `glob` in build.py could be used and added to main.swift.

**Comment by phausler** [comment]
Builds correctly now; running the tests on linux fail due to missing functionality in NSString; but overall the implementation looks good to me, I am going to merge this in and just disable the test until NSString implementations get up to par to deal with this.

**Comment by terhechte** [comment]
Awesome, thanks!

**Pull request merged**